### PR TITLE
Docs: Small plotting note for custom implementations

### LINKED ISSUE #1740: Add a `PlottableMetric` class
## 🚀 Feature

Add a `PlottableMetric` class, that will contain all the code related to metric plotting.

### Motivation

Currently, `plot` is an "abstract" method defined in `Metric`. To fully respect [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle), all `Metric` subclasses should give an implementation to `plot`. 

Most user-defined `Metric` sub-classes do not override `plot` (because they were coded prior to the `plot` feature). Also, some built-in `Metric` sub-classes, such as `BinaryGroupStatRates` , `MultilabelStatScores`, `CompositionalMetric`, `CatMetric`, etc., do not override `plot`.

### Pitch

A solution would be to remove the `plot` method (and related attributes) from `Metric`, and put them in a `PlottableMetric` class. Metrics that are plottable will simply extend both `Metric` and `PlottableMetric`. Metrics that should not be plottable will not extend `PlottableMetric`.

### Additional Context

I have not used the `plot` functionality yet, and I didn't work on it at all, so I'd be happy to learn more about the long-term goals for this `plot` feature (do you maybe plan to make all `Metric` sub-classes override it?)

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1740
Adds some details to the page on creating custom metric implementations on how to deal with the `plot` method.

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--1807.org.readthedocs.build/en/1807/

### CODE REVIEW

File: docs/source/pages/implement.rst [hunk 1]
```diff
@@ -48,21 +50,41 @@ Additionally you may want to set the class properties: `is_differentiable`, `hig
...
+        # Set to True if the metric during 'update' requires access to the global metric
+        # state for its calculations. If not, setting this to False indicates that all
+        # batch states are independent and we will optimize the runtime of 'forward'
+        full_state_update: bool = True
 
-   class MyMetric(Metric):
-      # Set to True if the metric is differentiable else set to False
-      is_differentiable: Optional[bool] = None
+Finally, from torchmetrics v1.0.0 onwards, we also support plotting of metrics through the `.plot` method. By default
+thie method will raise `NotImplementedError` but can be implemented by the user to provide a custom plot for the metric.
```
Comment (stancld) [inline docs/source/pages/implement.rst:69]:
```suggestion
this method will raise `NotImplementedError` but can be implemented by the user to provide a custom plot for the metric.
```

File: docs/source/pages/implement.rst [hunk 2]
```diff
@@ -48,21 +50,41 @@ Additionally you may want to set the class properties: `is_differentiable`, `hig
...
+For any metrics that returns a simle scalar tensor, or a dict of scalar tensors the internal `._plot` method can be
```
Comment (stancld) [inline docs/source/pages/implement.rst:70]:
```suggestion
For any metrics that returns a simple scalar tensor, or a dict of scalar tensors the internal `._plot` method can be
```

**Review by stancld — APPROVED** [review]

File: docs/source/pages/plotting.rst [hunk 3]
```diff
@@ -1,3 +1,5 @@
+.. _plotting:
```
Comment (Borda) [inline docs/source/pages/plotting.rst:1]: isn't it a bit confusing to have the page and label the same name?

**Review by Borda — APPROVED** [review]

File: docs/source/pages/plotting.rst (hunk 3 above)
Comment (SkafteNicki) [reply docs/source/pages/plotting.rst:1]: maybe, but we have the same thing multiple places in the docs

**Pull request merged**